cmd/kola/spawn.go

```diff
@@ -87,7 +91,23 @@ func doSpawn(cmd *cobra.Command, args []string) error {
 
 	var someMach platform.Machine
 	for i := 0; i < spawnNodeCount; i++ {
-		mach, err := cluster.NewMachine(userdata)
+		var mach platform.Machine
+		if kolaPlatform == "qemu" && spawnMachineOptions != "" {
+			b, err := ioutil.ReadFile(spawnMachineOptions)
+			if err != nil {
+				return fmt.Errorf("Could not read machine options: %v", err)
+			}
+
+			var machineOpts qemu.MachineOptions
+			err = json.Unmarshal(b, &machineOpts)
+			if err != nil {
+				return fmt.Errorf("Could not unmarshal machine options: %v", err)
+			}
+
+			mach, err = cluster.(*qemu.Cluster).NewMachineWithOptions(userdata, machineOpts)
+		} else {
+			mach, err = cluster.NewMachine(userdata)
```

Comment:
If you specify bad machine options (e.g. an invalid suffix) mach and err are both set to nil and it segfaults later.

Reply:
Forgot to setup the err var outside the if scope. Pushed a change to fix this.